### PR TITLE
Adds headless vault implementation to example app

### PR DESCRIPTION
**Pull request by HarryPrimer**

### CODE REVIEW

File: src/HeadlessUniversalCheckout/Managers/VaultManager.ts [hunk 1]
```diff
@@ -17,22 +17,23 @@ class PrimerHeadlessUniversalCheckoutVaultManager {
...
+            }
+        });
+    }
 
-    async fetchVaultedPaymentMethods(): Promise<PrimerVaultedPaymentMethod[]> {
+    async fetchVaultedPaymentMethods(): Promise<VaultedPaymentMethod[]> {
         return new Promise(async (resolve, reject) => {
             try {
-                const paymentMethods: PrimerVaultedPaymentMethod[] =
+                const data =
```
Comment (jnewc) [inline src/HeadlessUniversalCheckout/Managers/VaultManager.ts:34]: Is there a reason to lose the type information here?

File: src/HeadlessUniversalCheckout/Managers/VaultManager.ts [hunk 2]
```diff
@@ -50,20 +51,21 @@ class PrimerHeadlessUniversalCheckoutVaultManager {
...
-        })
+        });
     }
 
-    async validate(vaultedPaymentMethodId: String, additionalData: PrimerVaultedPaymentMethodAdditionalData): Promise<PrimerValidationError[]> {
+    async validate(vaultedPaymentMethodId: String, additionalData: VaultedPaymentMethodAdditionalData): Promise<ValidationError[]> {
         return new Promise(async (resolve, reject) => {
             try {
-                const errors: PrimerValidationError[] =
+                const data =
```
Comment (jnewc) [inline src/HeadlessUniversalCheckout/Managers/VaultManager.ts:60]: Same as above

File: example/android/app/build.gradle [hunk 3]
```diff
@@ -386,6 +386,7 @@ dependencies {
   implementation "io.primer:3ds-android:1.2.0"
   implementation "io.primer:ipay88-my-android:1.0.0"
   implementation "io.primer:klarna-android:1.0.1"
+  implementation "io.primer:nol-pay-android:1.0.1"
```
Comment (semirp) [inline example/android/app/build.gradle:389]: Is there a need to include this dependency?

File: example/src/screens/SettingsScreen.tsx [hunk 4]
```diff
@@ -24,13 +24,13 @@ export interface AppPaymentParameters {
     merchantName?: string;
 }
 
-export let customApiKey: string | undefined = "71347d48-c050-49c2-ac51-202525c4c0ec";
+export let customApiKey: string | undefined = "2b8537b2-478c-4315-88fb-f1954500b5fa";
```
Comment (semirp) [inline example/src/screens/SettingsScreen.tsx:27]: We should delete this :)

File: src/HeadlessUniversalCheckout/Managers/VaultManager.ts (hunk 1 above)
Comment (HarryPrimer) [reply src/HeadlessUniversalCheckout/Managers/VaultManager.ts:34]: Hey @jnewc Originally I was unwrapping the paymentMethods wrongly, I presumed that we would get a `PrimerVaultedPaymentMethod[]` array back but in actual fact we get an object with a `paymentMethods` field on it that contains the array.

File: src/HeadlessUniversalCheckout/Managers/VaultManager.ts (hunk 2 above)
Comment (HarryPrimer) [reply src/HeadlessUniversalCheckout/Managers/VaultManager.ts:60]: Hey @jnewc Originally I was unwrapping the PrimerValidationError wrongly, I presumed that we would get a `PrimerValidationError[]` array back but in actual fact we get an object with a `validationErrors ` field on it that contains the array.

File: example/android/app/build.gradle (hunk 3 above)
Comment (HarryPrimer) [reply example/android/app/build.gradle:389]: Hey @semirp when using `io.primer:android:2.19.1` on calling `PrimerHeadlessUniversalCheckout.current.start()` it was throwing a Failed Resolution of:Lio/primer/nolpay/api/PrimerNolPay when the `io.primer:nol-pay-android:1.0.1` dependency was not included in the `build.gradle`

File: example/src/screens/SettingsScreen.tsx (hunk 4 above)
Comment (HarryPrimer) [reply example/src/screens/SettingsScreen.tsx:27]: Agreed I'll remove it.

File: src/HeadlessUniversalCheckout/Managers/VaultManager.ts (hunk 1 above)
Comment (jnewc) [reply src/HeadlessUniversalCheckout/Managers/VaultManager.ts:34]: It feels like we're missing exposure of the result model then. I'd expect to see a `VaultedPaymentMethodResult` type or something like that, whether bridged from native or defined as a typescript type.

Comment (HarryPrimer) [reply src/HeadlessUniversalCheckout/Managers/VaultManager.ts]: Resolved in latest changes!

File: src/HeadlessUniversalCheckout/Managers/VaultManager.ts (hunk 2 above)
Comment (HarryPrimer) [reply src/HeadlessUniversalCheckout/Managers/VaultManager.ts]: Resolved in latest changes!

File: example/android/app/build.gradle (hunk 3 above)
Comment (HarryPrimer) [reply example/android/app/build.gradle]: I've now removed the Nol dependency and updated to the latest Android SDK version (2.19.2)

**Review by jnewc — APPROVED** [review]

**Review by semirp — APPROVED** [review]

**Pull request merged**